cmd/thanos/rule.go

```diff
@@ -529,12 +533,15 @@ func runRule(
 		storepb.RegisterStoreServer(s, store)
 
 		g.Add(func() error {
+			readinessProber.SetReady()
```

Comment:
For ruler it should behave the same as Prometheus. It has to be rdy ONLY when we start scraping.. Not sure how to ensure that. Is there any logic for this in `TSDB` we can reuse?

Reply:
What is the analogy here for scraping here? After first evaluation of rules?
I'll look at the prom and tsdb code how they do the readiness check.

Reply:
Looks like Prometheus just sets it's ready status right after the TSDB start

https://github.com/prometheus/prometheus/blob/f8932dde660161a9b390628eec617b41039c52c9/cmd/prometheus/main.go#L594

https://github.com/prometheus/prometheus/blob/f8932dde660161a9b390628eec617b41039c52c9/cmd/prometheus/main.go#L668

So possibly add it here?
https://github.com/improbable-eng/thanos/pull/656/files#diff-8b6a7e1ae18dc0ef5f768a537dff89f5R263

Reply:
> https://github.com/improbable-eng/thanos/pull/656/files#diff-8b6a7e1ae18dc0ef5f768a537dff89f5R263

Sorry this link does not work ): What place you mean?

Reply:
sorry, here right after the `tsdb.Open` finishes same way Prometheus does

https://github.com/improbable-eng/thanos/blob/master/cmd/thanos/rule.go#L263

Reply:
but now looking at the code that that the tsdb open is blocking on the start it should be ready only once the gRPC api is ready I suppose? there is no need to be ready for the evaluation of rules

Reply:
Moved to to the point when gRPS server is ready which is always after the tsdb start